internal/lookoutingesterv2/instructions/instructions.go

```diff
@@ -0,0 +1,553 @@
+			continue
+		}
+		ts := *event.Created
+		switch event.GetEvent().(type) {
+		case *armadaevents.EventSequence_Event_SubmitJob:
+			err = c.handleSubmitJob(queue, owner, jobset, ts, event.GetSubmitJob(), update)
+		case *armadaevents.EventSequence_Event_ReprioritisedJob:
+			err = c.handleReprioritiseJob(ts, event.GetReprioritisedJob(), update)
+		case *armadaevents.EventSequence_Event_CancelledJob:
+			err = c.handleCancelJob(ts, event.GetCancelledJob(), update)
+		case *armadaevents.EventSequence_Event_JobSucceeded:
+			err = c.handleJobSucceeded(ts, event.GetJobSucceeded(), update)
+		case *armadaevents.EventSequence_Event_JobErrors:
+			err = c.handleJobErrors(ts, event.GetJobErrors(), update)
+		case *armadaevents.EventSequence_Event_JobRunAssigned:
+			err = c.handleJobRunAssigned(ts, event.GetJobRunAssigned(), update)
+		case *armadaevents.EventSequence_Event_JobRunRunning:
+			err = c.handleJobRunRunning(ts, event.GetJobRunRunning(), update)
+		case *armadaevents.EventSequence_Event_JobRunSucceeded:
+			err = c.handleJobRunSucceeded(ts, event.GetJobRunSucceeded(), update)
+		case *armadaevents.EventSequence_Event_JobRunErrors:
+			err = c.handleJobRunErrors(ts, event.GetJobRunErrors(), update)
+		case *armadaevents.EventSequence_Event_JobDuplicateDetected:
+			err = c.handleJobDuplicateDetected(ts, event.GetJobDuplicateDetected(), update)
+		case *armadaevents.EventSequence_Event_CancelJob:
+		case *armadaevents.EventSequence_Event_JobRunLeased:
+		case *armadaevents.EventSequence_Event_ReprioritiseJobSet:
+		case *armadaevents.EventSequence_Event_CancelJobSet:
+		case *armadaevents.EventSequence_Event_ResourceUtilisation:
+		case *armadaevents.EventSequence_Event_StandaloneIngressInfo:
+		case *armadaevents.EventSequence_Event_JobRunPreempted:
+			log.Debugf("Ignoring event type %T", event)
+		default:
+			log.Warnf("Ignoring unknown event type %T", event)
+		}
+		if err != nil {
+			c.metrics.RecordPulsarMessageError(metrics.PulsarMessageErrorProcessing)
+			log.WithError(err).Warnf("Could not convert event at index %d.", idx)
+		}
+	}
+}
+
+func (c *InstructionConverter) handleSubmitJob(
+	queue string,
+	owner string,
+	jobSet string,
+	ts time.Time,
+	event *armadaevents.SubmitJob,
+	update *model.InstructionSet,
+) error {
+	jobId, err := armadaevents.UlidStringFromProtoUuid(event.GetJobId())
+	if err != nil {
+		c.metrics.RecordPulsarMessageError(metrics.PulsarMessageErrorProcessing)
+		return err
+	}
+
+	// Try and marshall the job Json. This shouldn't go wrong but if it does, it'c not a fatal error
+	// Rather it means that the json won't be available in the ui
+	var jobProto []byte
+	apiJob, err := eventutil.ApiJobFromLogSubmitJob(owner, []string{}, queue, jobSet, ts, event)
+	if err == nil {
+
+		jobProtoUncompressed, err := proto.Marshal(apiJob)
+		if err != nil {
+			log.Warnf("Couldn't marshall job %s in jobset %s as json.  %+v", jobId, jobSet, err)
```

Comment:
albin converted me to using `log.WithError(err).Warn(...)` rather than `log.Warnf("%+v", err)`